Find a functionally equivalent code:com.android.internal.util.AsyncChannel.disconnect:COMMENT
Method Modifier: public      internal    
Comment:/**
 * Disconnect
 */

@@ -1,6 +1,7 @@
 {
     if ((mConnection != null) && (mSrcContext != null)) {
         mSrcContext.unbindService(mConnection);
+        mConnection = null;
     }
     try {
         // Send the DISCONNECTED, although it may not be received
@@ -14,22 +15,25 @@
     // Tell source we're disconnected.
     if (mSrcHandler != null) {
         replyDisconnected(STATUS_SUCCESSFUL);
+        mSrcHandler = null;
     }
     // Unlink only when bindService isn't used
     if (mConnection == null && mDstMessenger != null && mDeathMonitor != null) {
         mDstMessenger.getBinder().unlinkToDeath(mDeathMonitor, 0);
+        mDeathMonitor = null;
     }
 }

